2020-02-Spring/0_arrays_matrix/README.md

```diff
@@ -26,21 +26,56 @@ In the style of:
 
 <a name="p1"/>
 
-### 1. PROBLEM 1 TODO :bug:
+### 1. Best Time to Buy and Sell Stock
 
-Source: TODO :bug:
+Source: [LeetCode](https://leetcode.com/problems/best-time-to-buy-and-sell-stock/)
 
 #### Scenario
 
-Problem Statement TODO :bug:
+Say you have an array of n elements representing stock prices.
+
+Price of the stock on the first day is the first element, second day is the second element and so on. 
+
+If you were only permitted to complete at most one transaction (i.e., buy one and sell one share of the stock), design an algorithm to find the maximum profit.
```

Comment:
Here, "one transaction" is a bit confusing, since I tend to think of a transaction as **either** _buying_ **or** _selling_. Leaving a suggestion to say "two transactions" -- feel free to come up with something better which explains you must buy and then sell the stock to achieve profit (or minimize loss).

```suggestion
If you were only permitted to complete at most two transactions (i.e., buy one and sell one share of the stock), design an algorithm to find the maximum profit.
```

Reply:
Seconded on 2 transactions